src/map/utils/charutils.cpp

```diff
@@ -766,6 +759,61 @@ namespace charutils
         luautils::OnGameIn(PChar, zoning == 1);
     }
 
+    void LoadSpells(CCharEntity* PChar)
+    {
+        // disable all spells
+        PChar->m_SpellList.reset();
+        PChar->m_EnabledSpellList.reset();
+
+        // Compile a string of all enabled expansions
+        std::vector<const char*> expansions {"COP", "TOAU", "WOTG", "ACP", "AMK", "ASA", "ABYSSEA", "SOA"};
+        std::string enabledExpansions;
+
+        uint8 count = 0;
+        for (const char* expan : expansions)
```

Comment:
for (auto&& expan : {"COP", "TOAU", "WOTG", "ACP", "AMK", "ASA", "ABYSSEA", "SOA"})